pkg/cluster/px/px.go

```diff
@@ -854,6 +824,139 @@ func (ops *pxClusterOps) deleteAllPXComponents() error {
+	ds := &apps_api.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      wiperName,
+			Namespace: pxDefaultNamespace,
+			Labels:    labels,
+		},
+		Spec: apps_api.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            wiperName,
+							Image:           pxNodeWiperImage,
+							ImagePullPolicy: corev1.PullAlways,
+							Args:            args,
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: &trueVar,
+							},
+							ReadinessProbe: &corev1.Probe{
+								InitialDelaySeconds: 30,
+								Handler: corev1.Handler{
+									Exec: &corev1.ExecAction{
+										Command: []string{"cat", "/tmp/px-node-wipe-done"},
+									},
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "etcpwx",
+									MountPath: "/etc/pwx",
+								},
+								{
+									Name:      "hostproc1",
+									MountPath: "/hostproc/1/ns",
+								},
+								{
+									Name:      "optpwx",
+									MountPath: "/opt/pwx",
+								},
+							},
+						},
+					},
+					RestartPolicy:      "Always",
+					ServiceAccountName: talismanServiceAccount,
+					Volumes: []corev1.Volume{
+						{
+							Name: "etcpwx",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/etc/pwx",
+								},
+							},
+						},
+						{
+							Name: "hostproc1",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/proc/1/ns",
+								},
+							},
+						},
+						{
+							Name: "optpwx",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/opt/pwx",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return ops.runDaemonSet(ds, pxNodeWiperTimeout)
+}
+
+func (ops *pxClusterOps) runDaemonSet(ds *apps_api.DaemonSet, timeout time.Duration) error {
+	err := ops.k8sOps.DeleteDaemonSet(ds.Name, ds.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	t := func() (interface{}, bool, error) {
+		_, err := ops.k8sOps.GetDaemonSet(ds.Name, ds.Namespace)
+		if err == nil {
+			return nil, true, fmt.Errorf("daemonset: [%s] %s is still present", ds.Namespace, ds.Name)
+		}
+
+		return nil, false, nil
+	}
+
+	if _, err := task.DoRetryWithTimeout(t, daemonsetDeleteTimeout, defaultRetryInterval); err != nil {
+		return err
+	}
+
+	ds, err = ops.k8sOps.CreateDaemonSet(ds)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("started daemonSet: [%s] %s", ds.Namespace, ds.Name)
+
+	err = ops.k8sOps.ValidateDaemonSet(ds.Name, ds.Namespace, timeout)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("validated successfull run of daemonset: [%s] %s", ds.Namespace, ds.Name)
```

Comment:
nit: may be we should review all the logs in talisman and make them pretty (start line with upper case. etc) if they are customer visible.

Reply:
done